mesonbuild/modules/unstable_wayland.py

```diff
@@ -47,10 +47,9 @@ def scan_xml(self, state, args, kwargs):
         if self.scanner_bin is None:
             # wayland-scanner from BUILD machine must have same version as wayland
             # libraries from HOST machine.
-            dep = self.interpreter.func_dependency(state.current_node, ['wayland-client'], {})
-            self.scanner_bin = state.find_program('wayland-scanner',
-                                                  for_machine=MachineChoice.BUILD,
```

Comment:
This PR switches from using the BUILD version of the program, to not specifying that and thus falling back on the HOST version.

This may or may not be desirable... technically if there is some difference in the generated output between versions, dropping this is more reliable. The cost is that after merging this, in a cross-compile we will need to have an exe_wrapper.

It's very much not clear to me when we choose to look up a native or a host tool anyway, so :shrug:

Reply:
The way it was done for GNOME tools is a bit of mix of both native and build machines:
1. Looks in cross file, not native file
2. Checks in native pkgconfig dep
3. Check in build machine PATH

Better be consistent in Wayland module.